webui/react/src/hooks/useModal/JupyterLab/useModalJupyterLab.tsx

```diff
@@ -23,38 +24,40 @@ const { Item } = Form;
 const STORAGE_PATH = 'jupyter-lab';
 const DEFAULT_SLOT_COUNT = 1;
 
-const settingsConfig: SettingsConfig = {
-  settings: [
-    {
+const settingsConfig: SettingsConfig<JupyterLabOptions> = {
+  applicableRoutespace: STORAGE_PATH,
+  settings: {
+    name: {
       defaultValue: '',
-      key: 'name',
       skipUrlEncoding: true,
-      type: { baseType: BaseType.String },
+      storageKey: 'name',
+      type: union([string, undefinedType]),
     },
-    {
+    pool: {
       defaultValue: '',
-      key: 'pool',
       skipUrlEncoding: true,
-      type: { baseType: BaseType.String },
+      storageKey: 'pool',
+      type: union([string, undefinedType]),
     },
-    {
+    slots: {
       defaultValue: DEFAULT_SLOT_COUNT,
-      key: 'slots',
       skipUrlEncoding: true,
-      type: { baseType: BaseType.Integer },
+      storageKey: 'slots',
+      type: union([number, undefinedType]),
     },
-    {
-      key: 'template',
+    template: {
+      defaultValue: '',
       skipUrlEncoding: true,
-      type: { baseType: BaseType.String },
+      storageKey: 'template',
+      type: union([string, undefinedType]),
```

Comment:
how come these types are change from string to string | undef?

Reply:
Were settings implicitly unioned with `undefined` in the old code? That is, did `BaseType.String` mean `string | undefined`?

This actually has a default (though a bad one) so it seems like it could just be `string`.

Reply:
`slot` above definitely seems like it isn't optional and should just be `number`.

Also just a minor point but we really should be using `X | null` for optional types instead of `X | undefined`.

Reply:
Why is it `?` in the settings type?

Reply:
(I see that this is a pre-existing issue, just wondering if you knew why)